### DISCUSSION TITLE
Licensing and contributions

### DISCUSSION THREAD

**Discussion by mrsimpson** [chunibyocola/sc-translator-crx#135] [General]
Dear @chunibyocola 
Your plug-in is really great and we'd like to use it at our company to connect to the corporate translator we already run. 
As part of that, we'd like to pre-configure and default some options so that users cannot make mistakes using it. 
Given the GPL-3 license you apply, we need to publish also those customizations. 
This is a bit tricky, but I doubt it would be useful for others. 

Also, there will be some minor improvements we'd like to add (such as RegExes for assigning particular domains to a translation engine). I'd much enjoy to contribute them, but in the past you already said you don't really want others to mess with you code ;-) )

Therefore, I'd like to know what would be your preferred way of consuming your software. 
Given the GPL-3, I'd much appreciate if you accept contributions OR if you dual-licensed the project with a non
-copyleft-license (such as MIT or Apache-2).

**Comment by chunibyocola** [comment]
> Given the GPL-3, I'd much appreciate if you accept contributions OR if you dual-licensed the project with a non
-copyleft-license (such as MIT or Apache-2).

Sorry, it is no. But you can push to your fork, without publishing to any stores, it shoulds meet the requirements of GPL-3.

> Also, there will be some minor improvements we'd like to add (such as RegExes for assigning particular domains to a translation engine). I'd much enjoy to contribute them, but in the past you already said you don't really want others to mess with you code ;-) )

As you can see, this production is providing to the users around the world, every features should consider about that. If your ideas are benefit for most of the users, I will be very willing to add such of features. You can draw a new issue to share your ideas, and I will deal with that as soon as I can : ) .

**Reply by mrsimpson** [reply]
> Sorry, it is no. But you can push to your fork, without publishing to any stores, it shoulds meet the requirements of GPL-3.

Yes, pushing my fork would be definitely meet GPL-3 requirements. It's a bit trickier to set up in a company though: Some officers get nervous only reading "GPL" and say "no, we don't do this in our company". I might have to do more administrative work to convince everyone.

Ideally, I'd not have to fork though but can sketch the ideas in such a way that you can implement them and everybody benefits. 
Many German users would benefit from #126 , btw.

**Reply by chunibyocola** [reply]
> Many German users would benefit from #126 , btw.

Yes, I had seen that awesome pull request. But I know no German, so that it is totally not maintainable for me, and I can't check how accurate it is. So I didn't close that PR and hope that PR will help German users.

**Reply by mrsimpson** [reply]
Well, translation is really not something to be enthusiastic about ;-)
But actually, lots of open source projects do i18n with the help of the community. There are online platforms (such as Lingohub) which are specialized just on that. 
I understand your level of perfection and you don't want your product to be imprecise at any place. 
However, "no native language" is definitely worse than "sometimes imprecise language". 
I'm a native German speaker and had a look at each single translation on the file. I could go through them once more (I was in fact unsure about one feature which makes it hard to translate) and ask another native German to review the result. Would that make you confident to merge it?
